lib/dangermattic/plugins/ios_release_check.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin for performing iOS / macOS release-related checks in a pull request.
+  #
+  # @example Checking for changes in Core Data models on a release branch:
+  #          ios_release_check.check_core_data_model_changed
+  #
+  # @example Checking for modified Localizable.strings on a regular branch:
+  #          ios_release_check.check_modified_localizable_strings
+  #
+  # @example Checking for synchronization between release notes and App Store strings:
+  #          ios_release_check.check_release_notes_and_app_store_strings
+  #
+  # @see Automattic/dangermattic
+  # @tags ios, macos, process, release
+  #
+  class IosReleaseCheck < Plugin
+    # Checks if an existing Core Data model has been edited in a release branch.
+    #
+    # @return [void]
+    def check_core_data_model_changed
+      warning = 'Do not edit an existing Core Data model in a release branch unless it hasn\'t been released to testers yet. ' \
+                'Instead create a new model version and merge back to develop soon.'
+
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { File.extname(path) == '.xcdatamodeld' },
+        message: warning,
+        on_release: true
+      )
+    end
+
+    # Checks if the Localizable.strings file has been modified on a regular branch, emiting a warning if that's the case.
+    #
+    # @return [void]
+    def check_modified_localizable_strings
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { path.end_with?('Resources/en.lproj/Localizable.strings') },
```

Comment:
`end_with?` is not an appropriate test to check paths, because that could match a partial name of a directory in the start of the string you're testing.

Besides, I'm not sure all our repos have their `en.lproj` inside `Resources`? (I think most do… but is that an absolute rule for _all_ our repos?

Finally, since this rule seems to be aimed at repositories that make use of `genstrings` (likely via [our `ios_generate_strings_file_from_code` action from `release-toolkit`](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb)), I think we should also warn if _any_ `Localizable.strings` file is modified, because:
 - The `en.lproj` file should only be updated when we run `genstrings` (on release branch, right after code freeze), but never updated manually
 - The other `*.lproj` files should only be updated when we download translations from GlotPress, which is also done on release branch, typically just before submission

Tumblr is a bit different, as it doesn't (yet) use `genstrings` to generate the `en.lproj` file (instead, developers update the `en.lproj/Localizable.strings` file manually when they need to add a new string). Yet, developers should not update any other `*.lproj/Localizable.strings` file manually themselves, as these translation files should only be updated on release branch when we download translations from OneSky.

All this means that:
 - For repos that use `genstrings`, neither `en.lproj` nor any other `*.lproj` file should be edited manually (but only updated by our automation on the release branch)
 - For Tumblr repo (or any other that don't use `genstrings`), editing `en.lproj/Localizable.strings` manually is allowed if on `develop` branch (or more precisely, if on any PRs which is NOT targeting `release/`), but editing any other `*.lproj/Localizable.strings` manually is not allowed, and should only be done via our automation on the `release/` branch

Reply:
Good points, thanks -- this will help to make the check a bit more general (it was first just based on what we had on Peril).
I've made a few changes here e687615ca59e7c026dd12188cec5df1413139559.
If I got it right, I've also added two separate checks to cover the "Tumblr case", 100847145854005297e37e78988d8848daad46ea and fefff99889dc406cd6a49f54c948acb5ee483d73.